lib/WebSocket.js

```diff
@@ -0,0 +1,144 @@
+module.exports = WebSocket;
+
+/**
+ * Dependencies.
+ */
+var Grammar =     require('./Grammar');
+var debug =       require('debug')('JsSIP:WebSocket');
+var debugerror =  require('debug')('JsSIP:ERROR:WebSocket');
+
+// 'websocket' module uses the native WebSocket interface when bundled to run
+// in a browser.
+var W3CWebSocket = require('websocket').w3cwebsocket;
+
+function WebSocket(url, node_ws_options) {
+  debug('new()');
+
+  var sip_uri = null;
+  var via_transport = null;
+
+  this.ws = null;
+
+  this.node_ws_options = node_ws_options || {};
```

Comment:
Well, we can write new code The New Way (TM), right? Then adjust the rest. Gradual changes are always easier to make than gigantic ones.